File: TelerikUIExamplesInCSharp/Examples/Chart/AxisTypes/NumericAxis.cs
using System;
using System.Collections.Generic;

using MonoTouch.Foundation;
using MonoTouch.UIKit;

using TelerikUI;

namespace Examples
{
	public class NumericAxis: ExampleViewController
	{
		public override void ViewDidLoad ()
		{
			base.ViewDidLoad ();

			TKChart chart = new TKChart (this.ExampleBounds);
			chart.AutoresizingMask = UIViewAutoresizing.FlexibleWidth | UIViewAutoresizing.FlexibleHeight;
			this.View.AddSubview (chart);

			Random r = new Random ();
			List<TKChartDataPoint> list = new List<TKChartDataPoint> ();
			for (int i = 0; i < 12; i++) {
				list.Add (new TKChartDataPoint (new NSNumber (i), new NSNumber (r.Next () % 2000)));
			}

			TKChartLineSeries series = new TKChartLineSeries (list.ToArray());
			series.SelectionMode = TKChartSeriesSelectionMode.Series;

			TKChartNumericAxis xAxis = new TKChartNumericAxis ();
			xAxis.Range = new TKRange (new NSNumber (0), new NSNumber (11));
			xAxis.Position = TKChartAxisPosition.Bottom;
			xAxis.MajorTickInterval = 1;
			chart.XAxis = xAxis;

			TKChartNumericAxis yAxis = new TKChartNumericAxis ();
			yAxis.Range = new TKRange (new NSNumber (0), new NSNumber (2000));
			yAxis.Position = TKChartAxisPosition.Left;
			yAxis.MajorTickInterval = 400;
			chart.YAxis = yAxis;

			chart.AddSeries (series);
		}
	}
}

File: TelerikUIExamplesInSwift2.0/TelerikUIExamplesInSwift/NumericAxis.swift
//
//  NumericAxis.swift
//  Swift Examples
//
//  Copyright (c) 2014 Telerik. All rights reserved.
//

import Foundation

class NumericAxis:ExampleViewController {
    
    let chart = TKChart()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        chart.frame = self.exampleBoundsWithInset
        chart.autoresizingMask = UIViewAutoresizing(rawValue:UIViewAutoresizing.FlexibleWidth.rawValue | UIViewAutoresizing.FlexibleHeight.rawValue)
        self.view.addSubview(chart)
        
        var array = [TKChartDataPoint]()
        for i in 0..<12 {
            array.append(TKChartBubbleDataPoint(x:i, y:Int(arc4random() % (2000))))
        }
        let series = TKChartLineSeries(items:array)
        series.selectionMode = TKChartSeriesSelectionMode.Series
        
        let xAxis = TKChartNumericAxis(minimum:0, andMaximum:11)
        xAxis.position = TKChartAxisPosition.Bottom
        xAxis.majorTickInterval = 1;
        chart.xAxis = xAxis
        
        let yAxis = TKChartNumericAxis(minimum:0, andMaximum:2000)
        yAxis.majorTickInterval = 400
        yAxis.position = TKChartAxisPosition.Left
        chart.yAxis = yAxis
        
        chart.addSeries(series)
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
}
